ourjapanlife/findadoc-server
# Validate inputs for `searchSubmission` filters and orderBy fields

Part of #257 


# What changed
1. Added a helper function `validateSubmissionSearchFilters` that:
- Prevents ordering by `id`.
- Prevents ordering by fields that do not exist.
- Limits the number of `fieldsToOrder` to a maximum of two fields.
- Places a minimum of 0 and a maximum of 1000 in the `limit` field (0 returns the default limit of 20).
- Requires at least one `spokenLanguage` property to be filled in.

# Testing instructions
1. Run this query in GraphQl Studio:
```
query GetSubmissions {
    submissions(filters: { 
        isUnderReview: false,
        limit: 5,
        orderBy: [
            {
                fieldToOrder: "healthcareProfessionalName", 
                orderDirection: desc 
            }
        ]
        }) {
        id
        googleMapsUrl
        healthcareProfessionalName
        isUnderReview
        isApproved
        isRejected
        createdDate
        updatedDate
        spokenLanguages {
            iso639_3
            nameEn
            nameJa
            nameNative
        }
    }
}
```
It should succeed.
2. Change `limit: 5` to `limit: -1` or `limit: 1001` and it should fail.
3. Add the filter `healthcareProfessionalName: " "` and it should fail.
